sycl/plugins/unified_runtime/ur/adapters/level_zero/usm.cpp

```diff
@@ -378,156 +608,108 @@ static ur_result_t USMFreeImpl(ur_context_handle_t Context, void *Ptr) {
   return UR_RESULT_SUCCESS;
 }
 
-void *USMMemoryAllocBase::allocate(size_t Size) {
-  void *Ptr = nullptr;
+static ur_result_t USMQueryPageSize(ur_context_handle_t Context, void *Ptr,
+                                    size_t *PageSize) {
+  ZeStruct<ze_memory_allocation_properties_t> AllocProperties = {};
+  ZE2UR_CALL(zeMemGetAllocProperties,
+             (Context->ZeContext, Ptr, &AllocProperties, nullptr));
+  *PageSize = AllocProperties.pageSize;
 
-  auto Res = allocateImpl(&Ptr, Size, sizeof(void *));
+  return UR_RESULT_SUCCESS;
+}
+
```

Comment:
I was looking at the tests that were failing and it seems that with page size query number of allocations don't match anymore. I think we could expose extra env variable to enable/disable page size query for those tests. 

@jandres742 do you think that would be ok? Something like this: https://github.com/igchor/llvm/commit/1b0d397840c6f103488b6ea13430e9b767b76299

It would be pretty tricky to update the tests to take the page size queries into consideration as the number of allocations depends on number of contexts created, etc.

Reply:
Ummm, we shouldn't need an environment variable. This should work, let me take a look

Reply:
Better not to introduce new env var, we could do something like what I did here on top of this branch.

https://github.com/jandres742/llvm/commit/852c204b99f1301f0ea45a7e6e0f027535e4fe75